solutions/CSES Movie Festival II.mdx

```diff
@@ -2,78 +2,60 @@
 id: cses-1632
 source: CSES
 title: Movie Festival II
-author: Shreyas Thumathy
+author: Shreyas Thumathy, Benjamin Qi
 ---
 
-<Warning title = "">
-		
-This problem is significantly easier after solving Movie Festival.
+<!-- In this problem, we're asked the maximum number of movie intervals $(start, end)$ we can cover using $k$ people. -->
 
-</Warning>
+<!-- ## Main Idea -->
 
-In this problem, we're asked the maximum number of movie intervals $(start, end)$ we can cover using $k$ people rather than $1$.
+The first step is the same as that of [Movie Festival](/silver/greedy-sorting#example---the-scheduling-problem); sort the movies in increasing order of end time. For each movie in order, we will assign it to one of the $K$ members to watch (or none of them).
 
-# Main Idea
+Keep track of the time each member finishes watching the latest movie that has been assigned to them in a sorted collection (e.g. a `multiset` in C++). Initially, the collection consists of $K$ zeroes.
 
-First, solve the one person variant of this problem. We can use the same approach and include multiple people instead of just one.
+For each movie in order, we can assign a member to watch it if there exists an element in the sorted collection less than or equal to the start time of the movie. If there are multiple such elements, choose the greatest one (the member who finished watching the latest movie assigned to him the latest). Increment the answer and update the collection accordingly.
 
-Whenever we come across a movie, we check if someone who previously watched a movie can watch this one too. If there are multiple people that can have this movie assigned, choose the latest one (that way if we encounter a movie that has an earlier start time, we can still assign that movie).
+<!-- If there are multiple members, choose the one who finishes watching the previous movie latest (that way, if we encounter a movie that has an earlier start time, we can still assign that movie). -->
 
-If the above condition doesn't hold and there are currently less than K people watching a movie, we can just assign a new person to this movie.
+<LanguageSection>
 
-We can keep a hold of the people watching movies by storing it in some sort of collection, a multiset is used below.
+<CPPSection>
 
-Implementation using a multiset in C++:
+## C++ Implementation
 
 ```cpp
-
 #include <iostream>
 #include <vector>
 #include <set>
 #include <algorithm>
 using namespace std;
 
-int n, k;
-vector<pair<int, int> > v(200005);
-
 int main() {
-	cin >> n >> k;
-	v.resize(n);
-	for (int i = 0; i < n; i++) {
-		// this will allow us to sort by end time.
+	int n, k; cin >> n >> k;
+	vector<pair<int, int>> v(n);
```

Comment:
Technically pair is in `#include<utility>`, though I don't know how cpp headers work. Including `iostream`, `vector`, and `set` all automatically allow use of `pair`s but `chrono` doesn't.